agpalm25/Epic_Exec_CS321_CORE_APP
# Database for the extracted files from the resume

### Changes Introduced
- Added functionality to handle resume uploads in the `/application` route.
- Integrated `extract_text_from_pdf` to extract fields from uploaded resumes.
- Mapped extracted data to the `ApplicantResume` model, including:
  - `name`
  - `education`
  - `experience`
  - `skills`
  - `certifications`
  - `summary`
- Established a relationship between `ApplicantResume` and `ApplicantInformation` using `student_id`.

### How to Test
1. Upload a sample PDF resume via the application form.
2. Verify that:
   - A new record is created in the `applicant_resume` table.
   - Extracted fields (e.g., name, education, skills) match the content of the uploaded resume.
   - The `applicant_resume` record is linked to the correct `applicant_information` record using `student_id`.

### Why This Change
- Streamlines the management of resume data by automating extraction and storage.
- Reduces manual effort in populating applicant details during the application process.
- Ensures consistency and accuracy in storing applicant data.

### Potential Impact
- Requires the `resume` field in the application form to be mandatory for processing.
- Only supports `.pdf` files; invalid file types or improperly formatted resumes may result in partial or missing data extraction.
